src/main/java/org/gridsuite/ds/server/service/parameters/impl/CurveGroovyGeneratorServiceImpl.java

```diff
@@ -39,13 +42,20 @@ public String generate(List<CurveInfos> curveInfosList) {
         // config root template
         ST curvesST = new ST(curvesTemplate);
 
-        // group curves info by equipmentId
+        // indexing equipment type by equipmentId
+        Map<String, EquipmentType> equipmentTypeByEquipmentIdMap = curveInfosList.stream()
+                .collect(Collectors.groupingBy(CurveInfos::getEquipmentId,
+                        Collectors.collectingAndThen(Collectors.toSet(), set -> set.stream().findFirst().get().getEquipmentType())));
+
+        // group variables by equipmentId
         Map<String, String> variablesByEquipmentIdMap = curveInfosList.stream()
-                .collect(Collectors.groupingBy(CurveInfos::getEquipmentId, Collectors.mapping(curveInfo -> "\"" + curveInfo.getVariableId() + "\"", Collectors.joining(", "))));
+                .collect(Collectors.groupingBy(CurveInfos::getEquipmentId, LinkedHashMap::new,
+                        Collectors.mapping(curveInfo -> "\"" + curveInfo.getVariableId() + "\"", Collectors.joining(", "))));
 
         String[] curveStringList = variablesByEquipmentIdMap.entrySet().stream().map(variablesByEquipmentId -> {
             ST curveST = new ST(curveTemplate);
-            curveST.add("dynamicModelId", "\"" + variablesByEquipmentId.getKey() + "\"");
+            curveST.add("idName", isStaticType(equipmentTypeByEquipmentIdMap.get(variablesByEquipmentId.getKey())) ? "staticId" : "dynamicModelId");
```

Comment:
```suggestion
            curveST.add("idType", isStaticType(equipmentTypeByEquipmentIdMap.get(variablesByEquipmentId.getKey())) ? "staticId" : "dynamicModelId");
```
Or I misunderstood ? Why `idName` ?

Reply:
In fact, i am generating a pair 'key value' for field 'id' in the template file, https://github.com/gridsuite/dynamic-simulation-server/pull/79/files#diff-2c7ff1d4283efb35791e70498385c45ea5e5e3dcf6bf41ae87064f098020217aR2

May be 'idKey' is better than 'idName' and 'idType'

Reply:
DONE by https://github.com/gridsuite/dynamic-simulation-server/pull/79/commits/4ca8d7423180964321314354ccdabf4627f3f363